src/back/TicketManager/src/PaymentService/TicketManager.PaymentService.Services/ApiKeyAuth/ApiKeyAuthorization.cs

```diff
@@ -0,0 +1,31 @@
+using FastEndpoints;
+using FluentValidation.Results;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using TicketManager.PaymentService.Services.Configuration;
+
+namespace TicketManager.PaymentService.Services.ApiKeyAuth;
+
+public class ApiKeyAuthorization<TRequest> : IPreProcessor<TRequest>
```

Comment:
I am not a fan of this solution. Any reason why you didn't use middleware? Also for the future don't work on two tasks when one of them is so complicated. You could have easilly just work on endpoint for status checking and make a new issue in Jira to work on `ApiKey`

Reply:
That's true, will remember for future tasks. As for the middleware, we don't use it here as I couldn't make it work. Just that. I couldn't make Attributes appear in Metadata which is needed to be able to attach the middleware only to the chosen set of endpoints. Hence, we decided with Michał that it is not that important to make it the greatest solution since it's only the current three endpoints that will be using them.